vulnerability/db.go

```diff
@@ -0,0 +1,117 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package vulnerability
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aquasecurity/trivy-db/pkg/metadata"
+	"github.com/elastic/elastic-agent-libs/logp"
+
+	"github.com/aquasecurity/trivy/pkg/commands/artifact"
+	"github.com/aquasecurity/trivy/pkg/flag"
+	trivy_types "github.com/aquasecurity/trivy/pkg/types"
+)
+
+type VulnerabilityDBManager struct {
+	log *logp.Logger
+}
+
+const (
+	dbRepository = "ghcr.io/aquasecurity/trivy-db"
+	cacheDir     = "/tmp/trivy-cache"
+	tmpCacheDir  = "/tmp/trivy-cache-tmp"
+)
+
+func NewVulnerabilityDBManager(log *logp.Logger) VulnerabilityDBManager {
+	log.Debug("NewVulnerabilityDBManager: New")
+
+	return VulnerabilityDBManager{
+		log: log,
+	}
+}
+
+func (f VulnerabilityDBManager) Download(ctx context.Context) error {
+	opts := flag.Options{
+		GlobalOptions: flag.GlobalOptions{
+			Timeout:  1 * time.Hour,
+			Quiet:    false,
+			Debug:    true,
+			CacheDir: tmpCacheDir,
+		},
+		VulnerabilityOptions: flag.VulnerabilityOptions{
+			VulnType:      []string{trivy_types.VulnTypeOS, trivy_types.VulnTypeLibrary},
+			IgnoreUnfixed: false,
+		},
+		ScanOptions: flag.ScanOptions{
+			Scanners: []trivy_types.Scanner{trivy_types.VulnerabilityScanner},
+			RekorURL: "https://rekor.sigstore.dev",
```

Comment:
what is this? can you please add some documentation?

Reply:
This is their default value, since we are skipping their config initialization (where trivy assigns default values) we need to set it ourselves.